### PR TITLE
Remove Bento helpers, 1st part

### PR DESCRIPTION
**Pull request by eduardoj**
We want to get rid of the helpers used only in the old UI.

For the sake of ease this pull request only takes in account helpers from `a` till `m` located under `src/api/app/helpers/webui/`.

Some refactoring was also done:  ~~move methods only used in its class to private, or~~ move a helper to its corresponding helper class.

### CODE REVIEW

**Comment by eduardoj** [comment]
[Remove bento helpers](https://trello.com/c/aaNWKJZf/1240-remove-bento-helpers)

**Comment by eduardoj** [comment]
Marking as "DO NOT MERGE" until fix spider tests are fixed.

**Comment by vpereira** [comment]
waiting #8220

File: src/api/app/helpers/webui/maintenance_incident_helper.rb [hunk 1]
```diff
@@ -100,6 +56,43 @@ def release_targets_cell(incident)
 
   private
```
Comment (coolo) [inline src/api/app/helpers/webui/maintenance_incident_helper.rb]: AFAICS rails does not support private functions in helpers - the nature of mixing modules into controllers plainly makes them private functions to every controller.

So IMO you just pollute git blame without much of a gain here.

Comment (eduardoj) [reply src/api/app/helpers/webui/maintenance_incident_helper.rb]: > AFAICS rails does not support private functions in helpers - the nature of mixing modules into controllers plainly makes them private functions to every controller.

Not what I wanted to achieve. I will remove that part. Thanks for the hint.

**Comment by eduardoj** [comment]
Spider tests are fixed! :tada: Removing "DO NOT MERGE" label.

File: src/api/app/helpers/webui/buildresult_helper.rb [hunk 2]
```diff
@@ -1,34 +1,4 @@
...
-
-      if enable_help && status['code']
-        concat(' ')
-        concat(sprite_tag('help', title: Buildresult.status_description(status['code'])))
-      end
-    end
-  end
-
   # NOTE: There is a JavaScript version of this method in project_monitor.js
   def webui2_arch_repo_table_cell(repo, arch, package_name, status = nil, enable_help = true)
```
Comment (hennevogel) [inline src/api/app/helpers/webui/buildresult_helper.rb]: What about renaming this one? :-)

Comment (eduardoj) [reply src/api/app/helpers/webui/buildresult_helper.rb]: Ops. I'll do it.

Comment (eduardoj) [reply src/api/app/helpers/webui/buildresult_helper.rb]: Done.

File: src/api/app/helpers/webui/monitor_helper.rb [hunk 3]
```diff
@@ -4,4 +4,15 @@ def self.print_statistics_array(array)
     array ||= []
     '[' + array.map { |time, value| "[#{time * 1000}, #{value}]" }.join(',') + ']'
   end
+
+  def icon_for_daemon(state)
```
Comment (hennevogel) [inline src/api/app/helpers/webui/monitor_helper.rb]: The commit message says you are moving `kiwi_image_breadcrumb` not this.

Comment (eduardoj) [reply src/api/app/helpers/webui/monitor_helper.rb]: Thanks! Fixed.

File: src/api/spec/helpers/webui/buildresult_helper_spec.rb [hunk 4]
```diff
@@ -1,100 +0,0 @@
-require 'rails_helper'
-
-RSpec.describe Webui::BuildresultHelper do
-  include Webui::WebuiHelper
-  let(:project) { create(:project) }
-  let(:package) { create(:package, project: project) }
-  let(:arch) { 'i586' }
-  let(:repo) { create(:repository, project: project, architectures: [arch]) }
-
-  describe '#arch_repo_table_cell' do
```
Comment (dmarcoux) [inline src/api/spec/helpers/webui/buildresult_helper_spec.rb]: This helper is still present. We should update these specs instead of removing them.

Comment (eduardoj) [reply src/api/spec/helpers/webui/buildresult_helper_spec.rb]: I will update the specs.

Comment (eduardoj) [reply src/api/spec/helpers/webui/buildresult_helper_spec.rb]: Updated. Please re-review.

**Comment by dmarcoux** [comment]
@eduardoj is not available today and tomorrow, so I will take over.

**Comment by dmarcoux** [comment]
@hennevogel Please have another look.

**Comment by dmarcoux** [comment]
@hennevogel Again please? :smile:

**Comment by dmarcoux** [comment]
I just realized that the `group_with_icon` helper is needed. I dropped the commit removing it. The helper is used in the partial which we need to migrate to Bootstrap. See the issue @eduardoj created: https://github.com/openSUSE/open-build-service/issues/8223.

**Pull request merged**